doc/Makefile

```diff
@@ -52,6 +52,8 @@ clean:
 
 UPLOAD_DIR=/srv/docs_scipy_org/doc/scipy-$(RELEASE)
 
+# `set -o pipefail` is specific to bash
+SHELL = /bin/bash
```

Comment:
`make dist` succeeds on this branch locally on the Linux box I use for doing releases/building docs, and without the `pipefail` complaint, so I'm +1. I did notice some suggestion to use `:=` for instant assignment in `Makefile`, but doesn't matter here I don't think.

We should merge if docs build passes in CI I think.